### ISSUE TITLE
Remove device compatibility check (to run on any android phone)

### ISSUE THREAD

**Issue by johanrosse** [dexcomapp/dexcomapp#27]
Please can you remove the device compatibility check (to run on any android phone).
Like on this app: https://docs.google.com/forms/d/e/1FAIpQLScD76G0Y-BlL4tZljaFkjlwuqhT83QlFM5v6ZEfO7gCU98iJQ/viewform

It would make it possible to have local broadcast to xdrip with android phones that is not officialy accepted by dexcom.

**Comment by ghost** [comment]
That would be great milestone for those, not having to buy a new device to use G6. And even a great  sustainable feature ;-)

**Comment by Ljuslykta** [comment]
Yes! Was just about to post this. Was in contact with the author of that app, it was just a matter of patching out two calls in the code. When digging around, I found that the probable suspect is in the files com/dexcom/cgm/appcompability and com/dexcom/cgm/activities where there's like 3 files called AppCombability. Tried doing it myself but since I'm no expert on Android I couldn't build the final file.

Anyway, is this app still developed here on Github or is it abandoned?

**Comment by efrost123** [comment]
I hope development is not stopped here :)
Let me try to see whether I can turn this off. I will be able to get to this over the weekend but I am really not sure how to publish it safely... Ping me if there are ideas about it.

**Comment by Ljuslykta** [comment]
I actually managed to install the patched version on my unsupported Mate 20 Pro , so I guess the check is already gone?

**Comment by efrost123** [comment]
I did not touch anything yet, but check might be skipped if there is no answer from server or no internet connection. This is my assumption.
Anyways, I believe I found piece of code that might disable compatibility check. I don't have not compatible phone to test though.

To all who wants to test - which region do you need?

**Comment by johanrosse** [comment]
I live in Sweden and have a Huawei p30.

**Comment by efrost123** [comment]
I assume it should be region 3 for Sweden. Uploaded file here: https://gofile.io/?c=GFbfry
zip password: dcehfdyj1974

What was changed:
in the com.dexcom.cgm.component_provider.c
original code:
```java
public boolean getIsAppCompatibilityEnabled() {
        return this.m_isAppCompatibilityEnabled;
    }
```
new code:
```java
public boolean getIsAppCompatibilityEnabled() {
        this.m_isAppCompatibilityEnabled = false;
        return this.m_isAppCompatibilityEnabled;
    }
```

**Comment by johanrosse** [comment]
I can not download: The upload has expired...

**Comment by efrost123** [comment]
Re-uploaded: https://gofile.io/?c=WURaJU
Will expire in one week. 
Archive password the same

Please, let me know whether it works

**Comment by johanrosse** [comment]
Now I can install and run the app. 
I have not tested to connect transmitter yet.
I am in middle in sensor session a d have 6 days left. Then I will connect with dexcom app.

**Comment by ghost** [comment]
> 
> 
> I did not touch anything yet, but check might be skipped if there is no answer from server or no internet connection. This is my assumption.
> Anyways, I believe I found piece of code that might disable compatibility check. I don't have not compatible phone to test though.
> 
> To all who wants to test - which region do you need?

Hey efrost123, Germany please, thanks.

**Comment by efrost123** [comment]
> > I did not touch anything yet, but check might be skipped if there is no answer from server or no internet connection. This is my assumption.
> > Anyways, I believe I found piece of code that might disable compatibility check. I don't have not compatible phone to test though.
> > To all who wants to test - which region do you need?
> 
> Hey efrost123, Germany please, thanks.

Any chance you can check region? Germany - probably region 3 and if so, use link above to try

**Comment by AlyxR** [comment]
Could anyone take look removing dexcom one compatibility check?
